python/flask/app-run-security-config.yaml

```diff
@@ -0,0 +1,53 @@
+
+# Flask check3: Flask application app.run best practices https://flask.palletsprojects.com/en/1.1.x/api/?highlight=run#flask.Flask.run
+rules:
+  # # . Rules for app.run() scope
+  - id: app_run_with_guard
+    pattern: |
+      if $X == '__main__':
+        app.run(...)
+    message: Running app.run(...) with guard is recommended
+    languages: [python]
+    severity: OK
+  - id: app_run_with_guard
+    pattern: |
+      if $X == '__main__':
+        ...
+        app.run(...)
+    message: Running app.run(...) with guard is recommended
+    languages: [python]
+    severity: OK
+  - id: app_run_with_guard_no_option
+    pattern: |
+      if $X == '__main__':
+        app.run()
```

Comment:
app.run(...) should match calls to app.run() with nothing.
... matches 0 or more arguments.

Reply:
Thanks for clarifying. I was not sure if `...` pattern did >0 or >1